### DISCUSSION TITLE
picked protein issue

### DISCUSSION THREAD

**Discussion by sdgroeve** [wfondrie/mokapot#84] [Q&A]
Hi,

I'm running mokapot with the picked protein method. I have a fasta file with one protein being a subset of another protein:

```
>O00762ups|UBE2C_HUMAN_UPS
HHHHHHMASQNHDPAATSVAAARKGAEPSGGAARGPVGKRIQQEIMTIMMSGDKGISAFPESDNIFKWVGTIHGAAGTVYEDIRYKISIEFPSGYPYNAPTVKFITPCYHPNVDTQGNICIDIIKEKWSAIYDVRTIIISIQSIIGEPNIDSPINTHAAEIWKNPTAFKKYIQETYSKQVTSQEP
>UBE2C_HUMAN
MASQNRDPAATSVAAARKGAEPSGGAARGPVGKRIQQEIMTIMMSGDKGISAFPESDNIFKWVGTIHGAAGTVYEDIRYKISIEFPSGYPYNAPTVKFITPCYHPNVDTQGNICIDIIKEKWSAIYDVRTIIISIQSIIGEPNIDSPINTHAAEIWKNPTAFKKYIQETYSKQVTSQEP
```

When I run mokapot it does not complain about the digest settings, so these are ok. 
Now, if I run mokapot with both these proteins in the fasta file then these two proteins are not in the mokapot.proteins.txt file. When I run mokapot with just one of these proteins it is listed as identified in the mokapot.proteins.txt file. Should these two proteins not just be put in one protein group that is indeed identified?

kind regards,
Sven

**Comment by wfondrie** [comment]
Hi @sdgroeve,

Can you share more details?

- What digestion settings are you using?
- Do you have decoy proteins in the FASTA file as well?

In general, two proteins are grouped into a protein group when one protein's peptides form a proper subset of anothers'. However, in this case, they are not using a standard tryptic digest: the first sequence will generate `HHHHHHMASQNHDPAATSVAAARK`, whereas the second will generate `MASQNRDPAATSVAAARK`.

**Comment by sdgroeve** [comment]
Hi Will,

I recreated the issue with a smaller dataset.
I attached the pin file and two fasta files. The concat_with_dupl.fasta contains the same proteins as concat.fasta, but with one protein copied (Q8TZL6_PYRFU as Q8TZL6_dup_PYRFU) and `HHHHHH`in front of it (as in the previous post). 

I run mokapot as:
`mokapot --decoy_prefix decoy_ --proteins concat_with_dupl.fasta --missed_cleavages 3 --min_length 7 --max_length 30 result.pin`

When I run this with `concat.fasta` then Q8TZL6_PYRFU is in `mokapot.proteins.txt`. With `concat_with_dupl.fasta` neither Q8TZL6_PYRFU or Q8TZL6_dup_PYRFU are in `mokapot.proteins.txt`. That can't be no?

[data.zip](https://github.com/wfondrie/mokapot/files/10121318/data.zip)

**Comment by sdgroeve** [comment]
Hi Will,

Even just copying an identified protein in the fasta file (so without adding the `HHHHHH`) none of the two proteins are in the result file.
